Gems/AWSGameLift/Code/AWSGameLiftClient/Tests/AWSGameLiftClientLocalTicketTrackerTest.cpp

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/std/smart_ptr/make_shared.h>
+
+#include <AWSGameLiftClientFixture.h>
+#include <AWSGameLiftClientLocalTicketTracker.h>
+#include <AWSGameLiftClientMocks.h>
+
+#include <Request/IAWSGameLiftInternalRequests.h>
+
+using namespace AWSGameLift;
+
+static constexpr const uint64_t TEST_RACKER_POLLING_PERIOD_MS = 100;
+static constexpr const uint64_t TEST_WAIT_BUFFER_TIME_MS = 10;
+static constexpr const uint64_t TEST_WAIT_MAXIMUM_TIME_MS = 10000;
```

Comment:
Suggest to run the tests for 100 times or more on your local machine using ctest to make sure that the maximum time is set properly.

Reply:
done, 10 seconds is a reasonable maximum time for single thread execution.